test/algorithm/copy_n.cpp

```diff
@@ -12,21 +12,26 @@
 #include <stl2/detail/algorithm/copy_n.hpp>
 #include <algorithm>
 #include "../simple_test.hpp"
+#include "../single_pass_array.hpp"
 
 namespace ranges = __stl2;
 
+//void test() requires <IndirectlyCopyable<I, O>>{
+
+//}
+
 int main() {
-	static const int source[] = {5,4,3,2,1,0};
-	static constexpr std::ptrdiff_t n = sizeof(source)/sizeof(source[0]);
+	single_pass_array source = {5,4,3,2,1,0};
+	static constexpr auto n = std::size(source);
 	int target[n];
 
 	std::fill_n(target, n, 0);
 	static_assert(n >= 2);
-	auto res = ranges::copy_n(source, n - 2, target);
-	CHECK(res.in == source + n - 2);
+	auto res = ranges::copy_n(source.begin(), n - 2, target);
+	CHECK(&(*res.in) == source.data() + n - 2);
```

Comment:
I know @CaseyCarter gets uncomfortable when seeing `&` used instead of `std::addressof`. I don't know if this is only in parameterised code, or if it's a general thing.

Reply:
It's about ADL-hijacking. `&x` can result in a call to some random overload of `operator&` when `x` has program-defined type. In this case, we know that `*res.in` is an lvalue `int`, so there's no problem using `&`.